nmdc_schema/migrators/migrator_from_11_7_0_to_11_8_0.py

```diff
@@ -0,0 +1,26 @@
+from nmdc_schema.migrators.migrator_base import MigratorBase
+
+class Migrator(MigratorBase):
+    r"""Migrates a database between two schemas."""
+
+    _from_version = "11.7.0"
+    _to_version = "11.8.0"
+
+    def upgrade(self) -> None:
+        r"""Migrates the database from conforming to the original schema, to conforming to the new schema."""
+        self.adapter.do_for_each_document("data_object_set", self.validate_data_object_type)
+
+    def validate_data_object_type(self, data_object: dict) -> None:
+        r"""
+        Raises an exception if the document either lacks a `data_object_type`
+        field or it has a `data_object_type` field whose value is falsey.
```

Comment:
Now that the condition has been updated, I recommend updating the docstring to match it. The current docstring reflects the previous behavior.

```suggestion
        Raises an exception if the document lacks a `data_object_type` field.
```